demo/frameworks/angular2/index.htm

```diff
@@ -0,0 +1,77 @@
+
+<head>
+    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/bootstrap-material-design/4.0.2/bootstrap-material-design.css" />
+    <link rel="stylesheet" href="../../common/index.css" />
+    <script src="https://npmcdn.com/core-js@2.4.1/client/shim.min.js"></script>
+    <script src="https://npmcdn.com/zone.js@0.8.12?main=browser"></script>
+    <script src="https://npmcdn.com/reflect-metadata@0.1.10"></script>
+    <script src="https://npmcdn.com/rxjs@5.4.1/bundles/Rx.min.js"></script>
+    <script src="https://npmcdn.com/@angular/core@4.2.3/bundles/core.umd.js"></script>
+    <script src="https://npmcdn.com/@angular/common@4.2.3/bundles/common.umd.js"></script>
+    <script src="https://npmcdn.com/@angular/compiler@4.2.3/bundles/compiler.umd.js"></script>
+    <script src="https://npmcdn.com/@angular/platform-browser@4.2.3/bundles/platform-browser.umd.js"></script>
+    <script src="https://npmcdn.com/@angular/platform-browser-dynamic@4.2.3/bundles/platform-browser-dynamic.umd.js"></script>
+    <script src="../../../dist/xcomponent.js"></script>
+    <script src="./login.js"></script>
+</head>
+
+<body>
+    <my-app>Loading...</my-app>
+    <script>
+
+        // component
+        (function(app) {
+            app.AppComponent =
+                ng.core.Component({
+                selector: 'my-app',
+                template: `
+                        <div>
+                            <h3>Log in on xyz.com</h3>
+                            <div id="container">
+                                <my-login-component [props]="props"></my-login-component>
```

Comment:
I thought we needed to support both method since both methods already supported in other drivers.

To be honest communicating between components through one (props) object is only natural to react people. Angular people (99% cases using typescript) would expose their props as public properties to component class and won't creat another object/type to hold props.


Related to that in angular functions are not passed down -technically possible but considered a bad practice and unnatural-. Instead event emitters emit events up and parent component handles it using an event binding: (OnLogin)="Login($event)"  Implementing that is in xcomponent is also possible and even having the two techniques configurable. 

I guess what I mean is providing natural and excpected model to people using a specific framework is more important that implementing all frameworks the same way.